src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -689,11 +689,12 @@ namespace Google.Example.Library.V1
     {
         /// <summary>
         /// A <see cref="BookName"/>-typed view over the <see cref="Name"/> resource name property.
+        /// May be set to <c>null</c>.
         /// </summary>
         public BookName BookName
         {
-            get { return BookName.Parse(Name); }
-            set { Name = GaxPreconditions.CheckNotNull(value, nameof(value)).ToString(); }
+            get { return string.IsNullOrEmpty(Name) ? null : Google.Example.Library.V1.BookName.Parse(Name); }
```

Comment:
I'm wondering whether Parse should just return null if we pass it "". It would simplify all the code calling it, but might feel a little odd. Let's think about that.